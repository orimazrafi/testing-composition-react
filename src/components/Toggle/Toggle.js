import React, { useState } from 'react'
export const Toggle = props => {
    const [display, setDisplay] = useState(true);
    const [number, setNumber] = useState(0);

    return (
        <div>
            <div>

                <button onClick={() => setDisplay(pre => !pre)} data-test="toggle-btn">Toggle</button>
                {display &&
                    <div data-test="toggled-element">Text To Show!</div>
                }
                <h1 data-test="number">

                    {number}
                </h1>
                <button data-test="increment-btn" onClick={() => setNumber(pre => pre + 1)}>increment</button>
            </div>

        </div>)
}