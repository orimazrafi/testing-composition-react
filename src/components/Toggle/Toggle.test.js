import React from 'react'
import { shallow } from "enzyme";
import { Toggle } from './Toggle';
import toJSON from "enzyme-to-json"
describe('Toggle', () => {
    it('should show text', () => {
        const wrapper = shallow(<Toggle />)
        const text = wrapper.find('[data-test="toggled-element"]');
        expect(text.text()).toBe("Text To Show!")
    });
    it('should hide text when button is clicked', () => {
        const wrapper = shallow(<Toggle />)
        const button = wrapper.find('[data-test="toggle-btn"]');
        button.simulate('click');
        const text = wrapper.find('[data-test="toggled-element"]');
        expect(text.length).toBe(0)
    })
})
describe('Increment', () => {
    it('should increment number when clicking the button.', () => {
        const wrapper = shallow(<Toggle />)
        const button = wrapper.find('[data-test="increment-btn"]');
        button.simulate("click");
        const text = wrapper.find('[data-test="number"]')
        expect(text.text()).toBe("1")
    });
    it('number should start with 0 value.', () => {
        const wrapper = shallow(<Toggle />)
        const text = wrapper.find('[data-test="number"]')
        expect(text.text()).toBe("0")
    });
    it('should hide text when button is clicked', () => {
        const wrapper = shallow(<Toggle />)
        const button = wrapper.find('[data-test="toggle-btn"]');
        button.simulate('click');
        const text = wrapper.find('[data-test="toggled-element"]');
        expect(text.length).toBe(0)
    })
})
describe("toggle snapshoot", () => {
    it('the snapshoot should not be change', () => {
        const wrapper = shallow(<Toggle />)
        expect(toJSON(wrapper)).toMatchSnapshot()
    })
})