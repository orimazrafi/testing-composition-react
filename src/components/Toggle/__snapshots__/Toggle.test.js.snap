// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle snapshoot the snapshoot should not be change 1`] = `
<div>
  <div>
    <button
      data-test="toggle-btn"
      onClick={[Function]}
    >
      Toggle
    </button>
    <div
      data-test="toggled-element"
    >
      Text To Show!
    </div>
    <h1
      data-test="number"
    >
      0
    </h1>
    <button
      data-test="increment-btn"
      onClick={[Function]}
    >
      increment
    </button>
  </div>
</div>
`;
